src/main/java/UserList.java

```diff
@@ -0,0 +1,35 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by mowall on 14/9/16.
+ */
+public class UserList implements GeneralList{
+    private List<User> users;
```

Comment:
also make the class level variables `final`.This will stop you from race conditons